homeassistant/components/smlight/diagnostics.py

```diff
@@ -0,0 +1,41 @@
+"""Collect diagnostics for SMLIGHT devices."""
+
+from __future__ import annotations
+
+import re
+from typing import Any
+
+from pysmlight import Api2
+from pysmlight.const import Actions
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.core import HomeAssistant
+
+from . import SmConfigEntry
+
+TO_REDACT = ["device_ip", "hostname", "MAC"]
+
+
+async def get_and_redact_log(client: Api2) -> list[str]:
+    """Get logs and redact sensitive data."""
+    log = await client.get({"action": Actions.API_GET_LOG.value})
+
+    mac_regex = r"([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})"
+    ip_regex = r"\b(?:[0-9]{1,3}\.){3}[0-9]{1,3}\b"
+    log = re.sub(f"{mac_regex}|{ip_regex}", "[REDACTED]", str(log))
```

Comment:
Why are we redacting mac and IP? I don't really see them as sensitive information. Yes, they can be used to explore ones network, but you need way more access to do that

Reply:
I thought they might be considered sensitive, but guess not. I will remove this.